Automattic/wp-calypso
# i18n: Pass WordPress.com links through localizeUrl

This is a follow-up to #27857

- [x] pass links to https://wordpress.com/tos/ through `localizeUrl()`
- [x] pass support links through `localizeUrl()`

#### Testing instructions
* Visit the components changed in this PR to make sure the URL reflects the right language version
* Test this with English and non-English UI (preferrably one Mag16 (e.g. `fr`) and one non-Mag16 (e.g. `pl`)
